huggingface_pipelines/speech.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from typing import Dict, Any, Optional
+from dataclasses import dataclass, replace
+from .pipeline import Pipeline, PipelineConfig, PipelineOverwrites
+import torch
+from sonar.inference_pipelines.speech import SpeechToTextModelPipeline
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class SpeechToTextOverwrites(PipelineOverwrites, total=False):
+    encoder_model: str
+    decoder_model: str
+    target_lang: str
+    pad_idx: int
+    fbank_dtype: str
+    n_parallel: int
+
+
+@dataclass
+class SpeechToTextPipelineConfig(PipelineConfig):
+    """
+    Configuration class for speech-to-text pipelines.
+    """
+    encoder_model: str = "sonar_speech_encoder_eng"
+    decoder_model: str = "text_sonar_basic_decoder"
+    target_lang: Optional[str] = None
+    pad_idx: int = 0
+    fbank_dtype: torch.dtype = torch.float32
+    n_parallel: int = 4
+
+    def with_overwrites(self, overwrites: SpeechToTextOverwrites):
+        return replace(self, **overwrites)
+
+
+@dataclass
+class HFSpeechToTextPipeline(Pipeline):
```

Comment:
why not decompose it into 2 : Speech to Embeddings and Embeddings to Text (which you already have)

Reply:
I wanted to leverage SpeechToTextPipeline from SONAR which will allow me to write less code, but I can decompose without an issue